### PR TITLE
cmd/clef, signer/core: password input fixes

### PR DESCRIPTION
**Pull request by holiman**
This PR uses a better terminal input for passwords, plus makes it possible to avoid boot-up warning. 
Before this PR, piping input would cause `inappropriate ioctl for device`, for some reason: 
```
yes only_for_demo | ./build/bin/clef --keystore pretixkeys/ newaccount

INFO [04-22|05:47:57.152] Starting clef                            keystore=pretixkeys/ light-kdf=false
DEBUG[04-22|05:47:57.152] FS scan times                            list=83.283µs set=1.699µs diff=2.31µs
## New account password

Please enter a password for the new account to be created (attempt 0 of 3)
> ERROR[04-22|05:47:57.152] Failed to read password                  err="inappropriate ioctl for device"
```
This PR uses instead the terminal from liner, already in use by the geth-codebase, and also makes it possible to avoid having to type `Ok` to the legal warning at boot up. 

Example 
```
[user@work clef]$ echo "only_for_test" | ./clef --keystore /tmp/foo newaccount --acceptWarn
INFO [04-22|11:03:38.815] Starting clef                            keystore=/tmp/foo light-kdf=false
DEBUG[04-22|11:03:38.815] FS scan times                            list=57.674µs set=13.758µs diff=6.428µs
## New account password

Please enter a password for the new account to be created (attempt 0 of 3)
!! Unsupported terminal, password will be echoed.
> 
-----------------------
DEBUG[04-22|11:03:39.999] FS scan times                            list=109.677µs set=32.022µs diff=7.499µs
INFO [04-22|11:03:40.203] Your new key was generated               address=0x2cCFc3444FBfbDfBc1C2AEe10a3275DD1756A02f
WARN [04-22|11:03:40.203] Please backup your key file!             path=/tmp/foo/UTC--2020-04-22T09-03-38.815872495Z--2ccfc3444fbfbdfbc1c2aee10a3275dd1756a02f
WARN [04-22|11:03:40.203] Please remember your password! 
Generated account 0x2cCFc3444FBfbDfBc1C2AEe10a3275DD1756A02f
```
cc @ligi

### CODE REVIEW

**Comment by holiman** [comment]
Turns out that my initial poc introduced circular dependencies. The prompter dragged in a pretty gigantic depedency-tree, since the same package also contains a lot of goja bindings and stuff. I now moved the (small) prompter into it's own package.

**Comment by ligi** [comment]
I can confirm that this solves the problem I had. Thanks so much for  addressing this issue so fast!

**Comment by ligi** [comment]
the only thing that is a bit weird is "!! Unsupported terminal, password will be echoed" - but then the password is not echoed (also not really sure why it should be)

**Comment by ligi** [comment]
@holiman added a small commit to mention the new cmdline param in the README - hope you don't mind

**Comment by holiman** [comment]
No, of course not, thanks a lot!

**Comment by holiman** [comment]
Although, in hindsight, it was a bad flag name. We don't usually camelCase flagNames, do we?

**Comment by ligi** [comment]
true - should be accept-warn

File: cmd/clef/README.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ GLOBAL OPTIONS:
    --stdio-ui              Use STDIN/STDOUT as a channel for an external UI. This means that an STDIN/STDOUT is used for RPC-communication with a e.g. a graphical user interface, and can be used when Clef is started by an external process.
    --stdio-ui-test         Mechanism to test interface between Clef and UI. Requires 'stdio-ui'.
    --advanced              If enabled, issues warnings instead of rejections for suspicious requests. Default off
+   --suppress-bootwarn     If set, does not show the warning during boot
```
Comment (fjl) [inline cmd/clef/README.md:49]: This option has a very weird name. How about `--no-warning`?

Comment (holiman) [reply cmd/clef/README.md:49]: okidok, will change

Comment (fjl) [reply cmd/clef/README.md:49]: We decided to not rename the flag after all.

**Pull request merged**